adafruit_dymoscale.py

```diff
@@ -0,0 +1,152 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2019 ladyada for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_dymoscale`
+================================================================================
+
+CircuitPython interface for DYMO scales.
+
+
+* Author(s): ladyada
+
+Implementation Notes
+--------------------
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+"""
+
+import time
+from pulseio import PulseIn
+from digitalio import DigitalInOut
+from micropython import const
+
+OUNCES = const(0x0B)   # data in weight is in ounces
+GRAMS = const(0x02)    # data in weight is in grams
+PULSE_WIDTH = 72.5
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_scale.git"
+
+class Scale:
+    """Interface to a DYMO postal scale.
+    """
+    def __init__(self, data_pin, units_pin, timeout=1.0):
+        """Sets up a DYMO postal scale.
+        :param ~pulseio.PulseIn data_pin: The data pin from the Dymo scale.
+        :param ~digitalio.DigitalInOut units_pin: The grams/oz button from the Dymo scale.
+        :param double timeout: The timeout, in seconds.
+        """
+        self.timeout = timeout
+        # set up the toggle pin
+        self.units_pin = DigitalInOut(units_pin)
+        self.units_pin.switch_to_output()
+        self.dymo = PulseIn(data_pin, maxlen=96, idle_state=True)
+        self.stable = None
+        self.units = None
+
+    @property
+    def weight(self):
+        """Weight in grams"""
+        weight = self.get_scale_data()
+        if self.units == OUNCES:
+            self.units = 'oz'
+        elif self.units == GRAMS:
+            self.units = 'g'
+        return weight
```

Comment:
convert it to grams no matter which one, so that weight is always consistantly unit'd

Reply:
Done!